src/edu/ucsd/library/xdre/harvesting/CilHarvesting.java

```diff
@@ -0,0 +1,615 @@
+        List<String> result = new ArrayList<>();
+        Object v = data.get(fieldName);
+        if (v != null && v.toString().length() > 0) {
+            if (!(v instanceof JSONArray)) {
+                result.add(v.toString());
+            } else {
+                Iterator<Object> it = ((JSONArray)v).iterator();
+                while (it.hasNext()) {
+                    result.add(it.next().toString());
+                }
+            }
+        }
+
+        return result;
+    }
+
+    /*
+     * Convert json key to the prefix for note technical details
+     * @param key
+     * @return
+     */
+    private String convertKeyToPrefix(String key) {
+        String value = key;
+        switch(key.toUpperCase()) {
+            case FieldMappings.SOURCE_KEY_RELATIONTOINTACTCELL:
+                value = "Relation to intact cell";
+                break;
+            case "ITEMTYPE":
+                value = "Item type";
+                break;
+            case "IMAGINGMODE":
+                value = "Imaging mode";
+                break;
+            case "PARAMETERIMAGED":
+                value = "Parameter imaged";
+                break;
+            case "SOURCEOFCONTRAST":
+                value = "Source of contrast";
+                break;
+            case "VISUALIZATIONMETHODS":
+                value = "Visualization methods";
+                break;
+            case "PROCESSINGHISTORY":
+                value = "Processing history";
+                break;
+            case "DATAQUALIFICATION":
+                value = "Data qualification";
+                break;
+            default:
+                value = key.substring(0, 1).toUpperCase() + key.substring(1).toLowerCase();
+                break;
+        }
+
+        return value;
+    }
+
+    /**
+     * Convert to CSV format for Excel InputStream
+     * @param jsonConvertSxlFile
+     * @return
+     * @throws Exception
+     */
+    public String toCSV(String jsonConvertSxlFile) throws Exception {
+        Record rec = nextRecord();
+        String recordId = rec.recordID();
+        Document rdfDoc = rec.toRDFXML();
+        Element root = rdfDoc.getRootElement();
+        // loop through to include all objects
+        while(rec != null) {
+            if (!rec.recordID().equals(recordId)) {
+                root.add(rec.toRDFXML().selectSingleNode("//dams:Object").detach());
+            }
+            rec = nextRecord();
+        }
+
+        File rdfFile = writeRdfContent(recordId, rdfDoc.asXML());
+        RDFExcelConvertor converter = new RDFExcelConvertor(rdfFile.getAbsolutePath(), jsonConvertSxlFile);
+        return converter.convert2CSV();
+    }
+
+    /**
+     * Write converted RDF/XML to file
+     * @param recordId
+     * @param rdf
+     * @return
+     * @throws FileNotFoundException
+     * @throws IOException
+     */
+    private File writeRdfContent(String recordId, String rdf) throws FileNotFoundException, IOException {
```

Comment:
After looking at this a few times I see that it's necessary to convert to RDF because I suspect that's how we typically get to our CSV format. So this is smart and a timesaver for you. I'm just curious about the performance here with writing out all these temporary rdf files. Would it be a pain to do the conversion from the rdfDoc in memory already? If so, then that's fine. I'm just curious and I also know you've thought about this a lot already :)

Reply:
@mcritchlow Yes, this is a good question. I think we can do it in memory but we may also want to store a copy of the converted RDF/XML so that we can download it for inspection if needed. That is what we've done for ExcelStream RDF preview.
Here we just write once for the batch of files while exporting the batch to CSV format, so it won't take much time. But it seems like I should change it to use a more general name like `metadata_converted_yyyy-mm-dd.csv` . What do you think?

Reply:
Ah, reviewing the results is a good point/justification for doing this. Given the complexity involved in the mapping, that's probably a good idea for sorting out any issues, especially with what will be an initially large first batch.

I think that filename convention makes good sense.

Reply:
@mcritchlow I've added commit https://github.com/ucsdlib/damsmanager/pull/295/commits/ef763c5c1d0003c6b120a2daa0936d82fa83c845 to rename the RDF file, which will be something like "cil_metadata_processed-yyyyMMddHHmm-rdf.xml" while exported.